### PR TITLE
python3 compat

### PR DESCRIPTION
**Pull request by richo**
This is enough to make `make tidy` work if you're using python3

There's definitely more stuff to do, but PR'ing now to avoid bitrot

### CODE REVIEW

File: src/etc/featureck.py [hunk 1]
```diff
@@ -78,8 +78,15 @@
         if not filename.endswith(".rs"):
             continue
 
+        if sys.version_info.major == 2:
+            _open = lambda f: open(f, 'r')
+        elif sys.version_info.major == 3:
+            _open = lambda f: open(f, 'r', encoding="utf-8")
```
Comment (alexcrichton) [inline src/etc/featureck.py]: How come this was necessary? Did this run into problems for some files?

**Comment by alexcrichton** [comment]
Thanks @richo! Unfortunately I don't think we can move completely to accepting python 3 until LLVM's build system supports it, but this certainly can't hurt!

File: src/etc/featureck.py (hunk 1 above)
Comment (richo) [inline src/etc/featureck.py]: it exploded on some of the innards of `Rc`, although I didn't dig into it super far.

I did a little reading, and the gist was that evidently python2 would lossily coerce invalid ascii characters, whereas python3 treats them as a hard error. I can dig up what I was reading if you'd like.

**Comment by richo** [comment]
Yup. It's not even clear to me that we'd want to, but a few projects I work on have been stung repeatedly by Ubuntu's default python switching, and I was waiting on a build so I figured I'd at least get `make tidy` working under py3.

File: src/etc/featureck.py (hunk 1 above)
Comment (alexcrichton) [inline src/etc/featureck.py]: Is there a way to open a file as UTF-8 in python2? All our source files should be valid UTF-8 I believe at least

Comment (richo) [inline src/etc/featureck.py]: Added a commit that uses the `codecs` module to do it.

**Comment by richo** [comment]
Actually that approach also works on python3. Removed the janky branch.

**Pull request merged**